hana/lib/HANAService.js

```diff
@@ -68,12 +68,13 @@ class HANAService extends SQLService {
           return dbc
         } catch (err) {
           if (isMultitenant) {
-            // REVISIT: throw the error and break retry loop
             // Stop trying when the tenant does not exist or is rate limited
-            if (err.status == 404 || err.status == 429)
-              return new Promise(function (_, reject) {
+            if (err.status == 404 || err.status == 429) {
+              if (cds.requires.db?.pool?.builtin) throw err
```

Comment:
Should that eventually be an internal server error? If yes, we have to change `err.status`